File: server/app/Models/Korong.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Korong extends Model
{
    use HasFactory;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'korong';

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'name',
        'address',
        'phone',
        'email',
        'description',
        'coordinates',
        'wali_korong_id',
    ];

    /**
     * Get the wali korong that owns this korong.
     */
    public function waliKorong(): BelongsTo
    {
        return $this->belongsTo(User::class, 'wali_korong_id');
    }
}

File: server/Models/Korong.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Models;
namespace server.Models
{
    public class Korong
    {
        [Key]
        public long Id { get; set; }

        [Required, MaxLength(255)]
        public string Name { get; set; } = null!;

        [Required, MaxLength(255)]
        public string Address { get; set; } = null!;

        [Required, MaxLength(50)]
        public string Phone { get; set; } = null!;

        [Required, MaxLength(255)]
        public string Email { get; set; } = null!;

        public string Description { get; set; } = null!;

        // FK
        public long WaliKorongId { get; set; }
        public WaliKorong WaliKorong { get; set; } = null!;

        public ICollection<KorongCoordinate> Coordinates { get; set; } = new List<KorongCoordinate>();

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
